From 2580592432a8cff5e627fff5d600dcf3ff26d230 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:06 -0400
Subject: [tools] perf/callchain: Add callchain_cursor_snapshot()

Message-id: <1410181752-25631-242-git-send-email-jolsa@redhat.com>
Patchwork-id: 91907
O-Subject: [PATCH RHEL7.1 BZ1134356 241/307] perf callchain: Add callchain_cursor_snapshot()
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit be1f13e30862ab6b0fffaecd556856a965cefa0c
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Sep 10 13:38:00 2012 +0900

description
===========
The callchain_cursor_snapshot() is for saving current status of the
callchain.  It'll be used to accumulate callchain information for each node.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/callchain.h b/tools/perf/util/callchain.h
index 24a53d5..8f84423 100644
--- a/tools/perf/util/callchain.h
+++ b/tools/perf/util/callchain.h
@@ -167,4 +167,13 @@ int fill_callchain_info(struct addr_location *al, struct callchain_cursor_node *
 
 extern const char record_callchain_help[];
 int parse_callchain_report_opt(const char *arg);
+
+static inline void callchain_cursor_snapshot(struct callchain_cursor *dest,
+          struct callchain_cursor *src)
+{
+ *dest = *src;
+
+ dest->first = src->curr;
+ dest->nr -= src->pos;
+}
 #endif /* __PERF_CALLCHAIN_H */
-- 
1.7.1